### ISSUE TITLE
Regression: Undoing the deletion of a photo nulls the texture on its UnlitMaterial

### ISSUE THREAD

**Issue by shiftyscales** [Frooxius/NeosPublic#1723]
## Describe the bug
If you delete a photo with the context menu, and undo, the photo will reappear as a white unlit quad. On investigating it, the StaticTexture2D has been unassigned/nulled from the UnlitMaterial.

## To Reproduce
- Take a photo (or spawn one from the inventory)
- Delete it
- Undo

## Expected behavior
The reference to the StaticTexture2D should be maintained between deletion and undo on the UnlitMaterial.

## Bug information (please complete the following information):
 - How often does it happen: Always
 - Does the bug persist after restarting Neos? Yes
 - Neos Version: 2021.2.4.348
 - Neos Platform: Windows
 - Regression: Yes. Unknown which exact build was last working, sorry. I highly suspect it's related to the change made in `2021.2.3.616`:
 ```
- Heavily optimized CPU and memory usage when destroying objects with asset preservation by reducing redundant walking of the data model for each element and optimizing other operations
-- This will significantly reduce hitches when destroying some complex objects (e.g. through the context menu)
```

## Additional context
This issue appears to occur regardless of if the photo is a local or neosdb asset.

## Reporters:
Tatsu Kimiero# 3109

**Comment by H3BO3** [comment]
I've noticed this recently too.

**Comment by TehTurk** [comment]
Third. Happened today at the meetup.

**Comment by Frooxius** [comment]
Fixed in 2021.2.7.1345, thanks!
